src/test/login.test.ts

```diff
@@ -0,0 +1,114 @@
+import * as request from 'supertest';
+import { expect } from 'chai';
+import { getRepository } from 'typeorm';
+import { User } from '../entity/User';
+import * as jwt from 'jsonwebtoken';
+
+async function graphqlPost(type: 'login' | 'createUser', variables: any, token?: string) {
```

Comment:
Aqui provavelmente você não precisa usar o createUser

Reply:
De qualquer forma, geralmente entre uma função que recebe um parâmetro pra se usar if e 2 funções, as 2 funções cotumam cumprir melhor o objetivo em termos de clareza. Até pq vc pode pegar o que se repete entre essas funções, se for o caso, e jogar numa terceira.